### ISSUE TITLE
[Python] Column names with spaces failing to work as identifiers

### ISSUE THREAD

**Issue by ned2** [duckdb/duckdb#8214]
### What happens?

When using the Python Relational API and a column name has a space, referencing it as an identifier tends to result in things breaking. Below is an example of this happening when using indexing syntax ( eg `relation['col']`).

I'm pretty sure I've seen it happen in another context; which I'll add when I come across it again. 

### To Reproduce

Given this context:
```python
with open("test.csv", "w") as f:
    f.write('boop,"hello world",hello_world\n')
    f.write('boop1,"quack quack",quack_quack\n')

rel = duckdb.read_csv("test.csv", header=True)
```

This works fine:
```python
rel["hello_world"]
```

But this throws an error:
```python
rel["hello world"]
```
    AttributeError: This relation does not contain a column by the name of 'hello world'

### OS:

Ubuntu 23.04

### DuckDB Version:

0.8.2.dev1559

### DuckDB Client:

Python

### Full Name:

Ned Letcher

### Affiliation:

Thoughtworks

### Have you tried this on the latest `master` branch?

- [X] I agree

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

**Comment by Tishj** [comment]
Thanks for the report,

I can reproduce the issue, but this reproduction example is not really narrow enough, as the issue you are running into is related to #8209 , the relation is not aware of the names of the result.

That being said, when we fix the reproduction example to actually target the issue, it still fails
```py
import duckdb

rel = duckdb.sql('select 42 as "hello world"')
res = rel["hello world"].fetchall()
print(res)
```

**Comment by ned2** [comment]
Oh yes good point! I had forgotten that the column name issue obscured the space character issue. I had meant to address that in my report, but forgot. Indeed, the `AttributeError` above is due to the column name not being initialised correctly as mentioned in #8209.

I had intended to address that by adding another query that just selected everything from the initial relation. your minimal example is much better!

**Comment by papssapa** [comment]
Ned,
Try adding the parameter normalize_names, it should solve your issue.
`rel = duckdb.read_csv("test.csv", header=True, normalize_names=TRUE)`

**Comment by ned2** [comment]
this was fixed by #8294  

thanks @Tishj !!
